source/isaaclab/isaaclab/utils/string.py

```diff
@@ -293,14 +296,15 @@ def resolve_matching_names_values(
         data: A dictionary of regular expressions and values to match the strings in the list.
         list_of_strings: A list of strings to match.
         preserve_order: Whether to preserve the order of the query keys in the returned values. Defaults to False.
+        require_all_keys_matched: Whether to require that all keys in the dictionary are matched. Defaults to True.
```

Comment:
The wording here seems off. Is it that all variables in `list_of_strings` have a match, or all keys in the dictionary have a string to which they got matched to?